packages/compass-e2e-tests/helpers/commands/select-connection-menu-item.ts

```diff
@@ -7,17 +8,35 @@ export async function selectConnectionMenuItem(
   itemSelector: string
 ) {
   const selector = Selectors.sidebarFavorite(favoriteName);
-  // It takes some time for the favourites to load
-  await browser.$(selector).waitForDisplayed();
 
-  // workaround for weirdness in the ItemActionControls menu opener icon
-  await browser.clickVisible(Selectors.ConnectionsTitle);
+  await browser.waitUntil(async () => {
+    console.log('.');
+    if (
+      await browser
+        .$(Selectors.sidebarFavoriteMenuButton(favoriteName))
+        .isDisplayed()
+    ) {
+      return true;
+    }
 
-  // Hover over an arbitrary other element to ensure that the second hover will
-  // actually be a fresh one. This otherwise breaks if this function is called
-  // twice in a row.
-  await browser.hover(`*:not(${selector}, ${selector} *)`);
-  await browser.hover(selector);
+    // It takes some time for the favourites to load
+    await browser.$(selector).waitForDisplayed();
+
+    // workaround for weirdness in the ItemActionControls menu opener icon
+    await browser.clickVisible(
+      TEST_MULTIPLE_CONNECTIONS
+        ? Selectors.SidebarHeader
+        : Selectors.ConnectionsTitle
+    );
+
+    // Hover over an arbitrary other element to ensure that the second hover will
+    // actually be a fresh one. This otherwise breaks if this function is called
+    // twice in a row.
+    await browser.hover(`*:not(${selector}, ${selector} *)`);
+
+    await browser.hover(selector);
+    return false;
```

Comment:
what does the return false mean?

Reply:
await browser.waitUntil(f) will run `f` again and again until it resolves `true`. False means retry. This is basically copy/pasted from a similar function select-favorites-menu-item. I think I'll just clean it up once the lists are merged or in one of the cleanup tickets. This is part of why I haven't added helpers for dealing with active connections - we'd have yet another copy of everything.